docs/docs/contribute/software/add-new-metric-provider.md

```diff
@@ -0,0 +1,140 @@
+     as an example.
+
+     Each of the three functions expects a string containing a float value in it.
+     But for example purposes we returned some of the data accessible in the function.
+
+     Below is an example of a placeholder provider implementation.
+
+      ```go
+        {% include "./assets/example-code/placeholder-code-example.go" %}
+      ```
+
+      > **Note** Refer to the documentation of the
+      > [KeptnMetric](../../reference/crd-reference/metric.md)
+      > and
+      > [Analysis](../../reference/crd-reference/analysis.md)
+      > resources
+      > to understand what data should be retrieved from the methods inputs to compute accurate results.
+
+4. **Instantiate the Provider** in the `providers.NewProvider` function
+   in the `metrics-operator/controllers/common/providers/provider.go` file.
+   add a case for the `KeptnPlaceholderProviderType`.
+   Instantiate the placeholder provider struct and return it.
+
+    ```go
+    // Inside the providers package
+
+    // NewProvider function
+    func NewProvider(providerType string, log logr.Logger, k8sClient client.Client) (KeptnSLIProvider, error) {
+        switch strings.ToLower(providerType) {
+        case KeptnPlaceholderProviderType:
+            return &placeholder.KeptnPlaceholderProvider{
+                Log:        log,
+                HttpClient: http.Client{},
+            }, nil
+        // Other cases...
+        }
+    }
+    ```
+
+5. **Update the validation webhook and crd config:** To update the validation webhook and crd config of the metrics operator.
+   Add the provider name next to last providers on this
+   [line](https://github.com/keptn/lifecycle-toolkit/blob/main/metrics-operator/api/v1beta1/keptnmetricsprovider_types.go#L29)
+   to look like this
+
+    `// +kubebuilder:validation:Pattern:=prometheus|dynatrace|datadog|dql|placeholder`.
+
+     In the metric-operator directory run `make manifests` to update the metrics-operator crd config
+     Then modify the helm chart and the helm chart crd validation to match the update in the metrics-operator crd config
+  
+6. **Add Test Cases:**
+     * Write a unit test to validate your implementation at the function level.
+       Unit tests ensure that individual
+       functions behave as expected and meet their functional requirements.
+
+     * Include a Chainsaw test to validate the behavior of Kubernetes resources managed by your code.
+       Chainsaw tests simulate real-world scenarios and interactions within a Kubernetes cluster, ensuring
+       the correctness of your Kubernetes configurations and deployments.
+
+         **NB:** For testing the metricsprovider a single KeptnMetric and KeptnMetricsProvider is needed
+
+        Below are the steps for adding an integration test.
+
+        * In the directory `test/chainsaw/testmetrics`, create a folder `keptn-metrics-validation` in our case.
```

Comment:
Why the test should be called validation? It seems like you want to validate something, but that's not true

Reply:
what do you suggest i call it?

Reply:
metrics-provider-placeholder

Reply:
Alright.